src/main/java/ledger/user_interface/ui_controllers/ImportTransactionsPopupController.java

```diff
@@ -56,7 +51,15 @@ public void initialize(URL fxmlFileLocation, ResourceBundle resources) {
             try {
                 selectTransactionsFile();
             } catch (Exception e) {
-                System.out.println("Error on transaction submission: " + e);
+                this.setupErrorPopup("Error on transaction submission: " + e);
+            }
+        });
+
+        this.submitTrxnFileBtn.setOnAction((event) -> {
+            try {
+                this.getScene().getWindow().hide();
```

Comment:
Would close not be better for this? If we hide we will keep creating new windows every time hit ImportTransaction (memory leak)